invenio_search/api.py

```diff
@@ -186,3 +179,54 @@ def _get_user_hash(self):
             alg.update(user_hash.encode('utf8'))
             return alg.hexdigest()
         return None
+
+
+class PrefixedIndexList(list):
+    """Custom list type for avoiding double prefixing."""
+
+    pass
+
+
+class RecordsSearch(BaseRecordsSearch):
+    """Example subclass for searching records using index prefixing."""
+
+    def __init__(self, **kwargs):
+        """Using PrefixedIndexList type to avoid double prefixing."""
+        # at object instantiation, kwargs['index'] is not defined.
+        # Elasticsearch-dsl-py re-instantiated the object at each search
+        # by cloning it and passing as kwargs the list of indices
+        # kwargs['index'] = ['index-name1', 'index-name2']
+        _index_param = kwargs.get('index', getattr(self.Meta, 'index', None))
+        if not isinstance(_index_param, PrefixedIndexList):
+            if isinstance(_index_param, (tuple, list)):
+                _prefixed_index_list = [
+                    prefix_index(app=current_app,
+                                 index=_index) for _index in _index_param]
+                kwargs.update({'index': _prefixed_index_list})
+            elif isinstance(_index_param, six.string_types):
+                _splitted_index = _index_param.split(',')
```

Comment:
I would also be nice to `.strip()` the values of the list to avoid problems with users passing stuff like `'records, deposits, users '`